value.h

```diff
@@ -164,16 +165,8 @@ static inline void truncval_to_flags(value_t *dst, match_flags flags)
 {
     assert(dst != NULL);
 
-    dst->flags.u64b &= flags.u64b;
```

Comment:
Yes, the zero-er is another function.

This reimplementation might indeed do 6 "undefined" bit hands, if you didn't init the whole int16 for some reason, but the important point is that I should be guaranteed to get the right result on the flags, because they are at the same position in both match_flags.

Then I'll read from the single flags when I need them.

This is type punning, as I wrote, but it's the "right way (TM)" and it's fully supported, AFAIK.

I'm not a huge expert of these fringe behaviours, but I believe this one is fully portable (and, of course, it works).